superset-frontend/src/components/Chart/DrillBy/DrillByModal.tsx

```diff
@@ -175,7 +195,47 @@ export default function DrillByModal({
         `}
       >
         {metadataBar}
-        <DrillByChart formData={updatedFormData} />
+        <div
+          css={css`
+            margin-bottom: ${theme.gridUnit * 6}px;
+            .ant-radio-button-wrapper-checked:not(.ant-radio-button-wrapper-disabled):focus-within {
+              box-shadow: none;
+            }
+          `}
+        >
+          <Radio.Group
+            onChange={({ target: { value } }) => {
+              setDrillBy(value);
+            }}
+            defaultValue={DrillByType.Chart}
+          >
+            <Radio.Button
+              value={DrillByType.Chart}
+              data-test="drill-by-chart-radio"
+            >
+              {t('Chart')}
+            </Radio.Button>
+            <Radio.Button
+              value={DrillByType.Table}
+              data-test="drill-by-table-radio"
+            >
+              {t('Table')}
+            </Radio.Button>
+          </Radio.Group>
+        </div>
+        {drillBy === DrillByType.Chart && chartDataResult && (
```

Comment:
I think we shouldn't check for `chartDataResult` here because then we won't be displaying the loading spinner from `DrillByChart`. So we should either remove that check here or move the loading spinner to this component

Reply:
We probably should display the spinner for Table mode too

Reply:
i agreed =. i will move the spinner to the modal component